src/main/java/beam/physsim/jdeqsim/AgentSimToPhysSimPlanConverter.java

```diff
@@ -116,6 +117,58 @@ private void preparePhysSimForNewIteration() {
         jdeqsimPopulation = PopulationUtils.createPopulation(agentSimScenario.getConfig());
     }
 
+    /**
+     * Logs all the broken links (if any) observed with a path inside a network route.
+     */
+    private void checkForBrokenLinksInsideNetwork(MutableScenario jdeqSimScenario) {
+        List<String> brokenLinks = new ArrayList<>();
+        Map<Id<Link>, ? extends Link> links = jdeqSimScenario.getNetwork().getLinks();
+        jdeqSimScenario.getPopulation().getPersons().values().
+                // for each selected plan
+                forEach(f -> f.getSelectedPlan().getPlanElements().forEach(pe -> {
+                    //for each leg
+                    if(pe instanceof Leg && ((Leg) pe).getRoute() != null) {
+                        Leg leg = (Leg) pe;
+                        NetworkRoute networkRoute = (NetworkRoute) leg.getRoute();
+                        List<Id<Link>> inBetweenLinks = networkRoute.getLinkIds();
+                        ArrayList<Link> routeLinks = new ArrayList<>();
+                        //collect all the start + in between + end links that build the route
+                        routeLinks.add(links.get(leg.getRoute().getStartLinkId()));
+                        inBetweenLinks.forEach(id -> routeLinks.add(links.get(id)));
+                        routeLinks.add(links.get(leg.getRoute().getEndLinkId()));
+                        // check if there are any break in the links in the route
+                        for(int i = 0; i < routeLinks.size()-1;i++){
+                            if(routeLinks.get(i).getToNode() != routeLinks.get(i+1).getFromNode()){
+                                brokenLinks.add(routeLinks.get(i).getId().toString() + "->" + routeLinks.get(i+1).getId().toString());
+                            }
+                        }
+                    }
+                }));
+        if(!brokenLinks.isEmpty()) {
+            StringBuilder brokenPaths = new StringBuilder();
+            brokenLinks.forEach(l -> brokenPaths.append(l).append(","));
+            log.info("Broken link paths observed in network route : " + brokenPaths.toString());
+        }
+    }
+
+    /**
+     * Logs all the erroneous links that have same from and to nodes.
+     */
+    private void checkForErroneousLinkNodes() {
+        List<String> errorLinks = new ArrayList<>();
+        Collection<? extends Link> links = agentSimScenario.getNetwork().getLinks().values();
+        for (Link link : links) {
+            if(link.getFromNode() == link.getToNode()){
+                errorLinks.add(link.getId().toString());
+            }
+        }
+        if(!errorLinks.isEmpty()) {
+            StringBuilder errorLinkIds = new StringBuilder();
+            errorLinks.forEach(id -> errorLinkIds.append(id).append(","));
+            log.info("Links that have same from and to nodes: " + errorLinkIds.toString());
```

Comment:
just print number of links with same from and to node, linkIds can be printed with debug level

Reply:
done